### PR TITLE
MULE-18454: Cannot serialize artifact declaration using the managed oAuth connection provider

### PR DESCRIPTION
**Pull request by SebaElizalde**

### CODE REVIEW

File: modules/extensions-spring-support/src/test/java/org/mule/runtime/module/extension/internal/resources/OCSExtensionModelJsonGeneratorTestCase.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+import org.mule.test.oauth.TestOAuthExtension;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+
+import org.junit.runners.Parameterized;
+
+public class OCSExtensionModelJsonGeneratorTestCase extends ExtensionModelJsonGeneratorTestCase {
+
```
Comment (ndinu) [inline modules/extensions-spring-support/src/test/java/org/mule/runtime/module/extension/internal/resources/OCSExtensionModelJsonGeneratorTestCase.java:24]: The property needs to also be present for the deserialization test.

File: modules/extensions-spring-support/src/test/java/org/mule/runtime/module/extension/internal/resources/OCSExtensionModelJsonGeneratorTestCase.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  private static ExtensionModelLoader javaLoader = new DefaultJavaExtensionModelLoader();
+
+  @Parameterized.Parameters(name = "{1}")
+  public static Collection<Object[]> data() {
+    List<ExtensionJsonGeneratorTestUnit> extensions =
+        Collections.singletonList(newTestUnit(javaLoader, TestOAuthExtension.class, "test-oauth-ocs.json"));
+    try {
+      return SystemProperty.runWithProperty(OCS_ENABLED, "true", () -> createExtensionModels(extensions));
```
Comment (ndinu) [inline modules/extensions-spring-support/src/test/java/org/mule/runtime/module/extension/internal/resources/OCSExtensionModelJsonGeneratorTestCase.java:32]: Use static import.

File: tests/unit/src/main/java/org/mule/tck/junit4/rule/SystemProperty.java [hunk 3]
```diff
@@ -6,13 +6,33 @@
...
  * Sets up a system property before a test and guarantees to tear it down afterward.
  */
 public class SystemProperty extends ExternalResource {
 
+  /**
+   * Utility method to execute a callable with a system property and restoring the property's original value afterwards.
+   *
+   * @return the return value of the callable, or null if an exception was found during its execution.
+   */
+  public static <V> V runWithProperty(String name, String value, Callable<V> runnable) throws Throwable {
```
Comment (ndinu) [inline tests/unit/src/main/java/org/mule/tck/junit4/rule/SystemProperty.java:23]: If you need a Callable, rename variable and rename the method.

File: tests/unit/src/main/java/org/mule/tck/junit4/rule/SystemProperty.java [hunk 4]
```diff
@@ -6,13 +6,33 @@
...
+    V callableValue;
+    SystemProperty property = new SystemProperty(name, value);
+    property.before();
+    try {
+      callableValue = runnable.call();
+    } catch (Exception e) {
+      callableValue = null;
+    }
+    property.after();
```
Comment (ndinu) [inline tests/unit/src/main/java/org/mule/tck/junit4/rule/SystemProperty.java:32]: I would call the after method in a finally block, and throw and exception if one arises.

File: tests/unit/src/main/java/org/mule/tck/junit4/rule/SystemProperty.java [hunk 5]
```diff
@@ -6,13 +6,33 @@
...
 public class SystemProperty extends ExternalResource {
 
+  /**
+   * Utility method to execute a callable with a system property and restoring the property's original value afterwards.
+   *
+   * @return the return value of the callable, or null if an exception was found during its execution.
+   */
+  public static <V> V runWithProperty(String name, String value, Callable<V> runnable) throws Throwable {
+    V callableValue;
+    SystemProperty property = new SystemProperty(name, value);
```
Comment (ndinu) [inline tests/unit/src/main/java/org/mule/tck/junit4/rule/SystemProperty.java:25]: Seems weird to create a SystemProperty to only call the before and after method.

File: modules/extensions-spring-support/src/test/java/org/mule/runtime/module/extension/internal/resources/OCSExtensionModelJsonGeneratorTestCase.java (hunk 1 above)
Comment (SebaElizalde) [reply modules/extensions-spring-support/src/test/java/org/mule/runtime/module/extension/internal/resources/OCSExtensionModelJsonGeneratorTestCase.java:24]: What do you mean?

Comment (ndinu) [reply modules/extensions-spring-support/src/test/java/org/mule/runtime/module/extension/internal/resources/OCSExtensionModelJsonGeneratorTestCase.java:24]: Ignore this comment. I was wrong, I meant the load test, but it is not necessary.

File: modules/extensions-spring-support/src/test/java/org/mule/runtime/module/extension/internal/resources/OCSExtensionModelJsonGeneratorTestCase.java [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+    } catch (Throwable throwable) {
+      return null;
```
Comment (ndinu) [inline modules/extensions-spring-support/src/test/java/org/mule/runtime/module/extension/internal/resources/OCSExtensionModelJsonGeneratorTestCase.java:34]: Fail if this happens.

Comment (SebaElizalde) [reply modules/extensions-spring-support/src/test/java/org/mule/runtime/module/extension/internal/resources/OCSExtensionModelJsonGeneratorTestCase.java:34]: fixed

**Pull request merged**